internal/services/cognitive/cognitive_account_resource.go

```diff
@@ -324,6 +331,31 @@ func resourceCognitiveAccount() *pluginsdk.Resource {
 				Sensitive: true,
 			},
 		},
+
+		CustomizeDiff: func(ctx context.Context, d *pluginsdk.ResourceDiff, i interface{}) error {
+			if d.Get("project_management_enabled").(bool) && d.Get("kind").(string) != "AIServices" {
+				return fmt.Errorf("`project_management_enabled` can only be enabled when kind is set to `AIServices`")
+			}
+
+			if d.Get("project_management_enabled").(bool) && len(d.Get("identity").([]interface{})) == 0 {
+				return fmt.Errorf("for `project_management_enabled` to be enabled, a managed identity must be assigned. Please set `identity` to at least one SystemAssigned or UserAssigned identity")
+			}
+
+			if d.Get("dynamic_throttling_enabled").(bool) && utils.SliceContainsValue([]string{"OpenAI", "AIServices"}, d.Get("kind").(string)) {
+				return fmt.Errorf("`dynamic_throttling_enabled` is currently not supported when kind is set to `OpenAI` or `AIServices`")
+			}
+
+			if d.HasChange("project_management_enabled") && !d.Get("project_management_enabled").(bool) {
+				return fmt.Errorf("once `project_management_enabled` is enabled, it cannot be disabled")
+			}
+
+			if d.HasChange("customer_managed_key") && d.Get("project_management_enabled").(bool) {
```

Comment:
This might also cause issue when users update from MS managed key to customer managed key?

Reply:
The absence of `customer_managed_key` is called MS managed.